icu4c/source/common/unicode/uchar.h

```diff
@@ -3110,6 +3118,12 @@ u_isxdigit(UChar32 c);
 U_CAPI UBool U_EXPORT2
 u_ispunct(UChar32 c);
 
+U_CAPI UBool U_EXPORT2
```

Comment:
u_isOpenPunct  / u_isClosePunct  are only used inside common/rbbi.cpp. I do not believe we need to add a new API for that unless you like to bring an API proposal to ICU TC for that. The other way is you just move these code into common/rbbi.cpp as a internal function in that file . Same to U_GC_OP_MASK / U_GC_CP_MASK

Reply:
Fixed